federation/pkg/federatedtypes/adapter.go

```diff
@@ -54,6 +54,15 @@ type FederatedTypeAdapter interface {
 	ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error)
 
 	NewTestObject(namespace string) pkgruntime.Object
+
+	// ImplementsReconcileHook is used to explicitly state, if the adapter implements ReconcilePlugin interface also.
+	ImplementsReconcilePlugin() bool
```

Comment:
I would prefer if you could separate this and all related code into a separate PR.  It should be fairly trivial, and very easy to understand and/or revert if necessary.